File: sort/pro_5.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int nn=0;

bool compare(string s1, string s2){
    if(s1[nn]>s2[nn])   return false;
    else if (s1[nn]<s2[nn]) return true;
    else return s1<s2;
}

vector<string> solution(vector<string> strings, int n) {
    nn=n;
    sort(strings.begin(),strings.end(),compare);
    return strings;
}

File: dp/pro_5.java
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] money) {
        Arrays.sort(money);
        
        int [] dp = new int[n+1];
        for(int i=0; i<=n; i++){
            if(i%money[0]==0)   dp[i]=1;
            else dp[i]=0;
        }
        
        for(int i=1; i<money.length; i++){
            for(int j=money[i];j<=n; j++){
                dp[j]+=dp[j-money[i]];
                dp[j]%=1000000007;
            }
        }
        
        return dp[n]%1000000007;
    }
}